### PR TITLE
[Libmamba] Remove code that now belongs to `conda-incubator/conda-libmamba-solver`

### PR DESCRIPTION
**Pull request by jaimergp**
Sister PR for https://github.com/conda-incubator/conda-libmamba-solver/pull/1

This should leave `poc-libsolv` with minimal changes. Just the minimum amount necessary to make the new solver logic work.

### CODE REVIEW

**Comment by jaimergp** [comment]
All yours @jezdez

File: conda/testing/helpers.py [hunk 1]
```diff
@@ -20,18 +20,18 @@
 from time import time
```
Comment (jezdez) [inline conda/testing/helpers.py:20]: Those moves make me happy!

File: conda/testing/solver_helpers.py [hunk 2]
```diff
@@ -0,0 +1,1280 @@
+from __future__ import annotations
```
Comment (jezdez) [inline conda/testing/solver_helpers.py:1]: Not sure if the overlap of `conda/testing/solver_helpers.py` with `conda/testing/helpers.py` is useful, maybe worth doing a sub-package like `conda.testing.helpers.solvers` or simply `conda.testing.solvers`?

File: dev/windows/setup.bat [hunk 3]
```diff
@@ -1,9 +1,17 @@
 @ECHO ON
-pushd %TEMP% || goto :error
+:: Obtain abs path to the conda source root
+pushd "%~dp0"
+    set "TWO_DIRS_UP=..\..\"
+    set "CONDA_SRC_PATH="
+    pushd %TWO_DIRS_UP% || goto :error
+        set "CONDA_SRC_PATH=%CD%"
+    popd || goto :error
+popd || goto :error
```
Comment (jezdez) [inline dev/windows/setup.bat:9]: 🤯

**Review by jezdez — APPROVED** [review]
Only minor nits! 💃🏻

File: dev/windows/setup.bat (hunk 3 above)
Comment (jaimergp) [reply dev/windows/setup.bat:9]: Yeah I know, at least we can have visual indentation to see what's going on 😂

File: conda/testing/solver_helpers.py (hunk 2 above)
Comment (jaimergp) [reply conda/testing/solver_helpers.py:1]: I was hesitating whether to merge or not, but given the history of that file (and how radically different the approaches are; @FFY00 did an AWESOME work with that test interface), I decided against that. 

Let me know how you want to proceed here:

* Leave files separate but rename/move to make it more clear
* Perform surgery on `conda.testing.helpers` and move things that concern solvers to `conda.testing.solver`

Comment (jaimergp) [reply conda/testing/solver_helpers.py:1]: Can we tackle this in a different PR?

Comment (jezdez) [reply conda/testing/solver_helpers.py:1]: Sounds good!

File: conda/testing/helpers.py (hunk 1 above)
Comment (jaimergp) [reply conda/testing/helpers.py:20]: We need to clean some unused symbols but I'll do that on a coming PR!

**Comment by jaimergp** [comment]
Ok, I think we can clean a bit more, but we'll get a clearer picture once this is merged and we can compare against master directly.

Ready to merge @jezdez !

**Pull request merged**